File: LeetCode/DataStruct/Golang/Leetcode946-validate-stack-sequences.go
package sbgo

func validateStackSequences(pushed []int, popped []int) bool {
  ans,n := []int{},len(pushed)
  for i,j := 0,0;i < n;i++ {
    ans = append(ans,pushed[i])
    for len(ans) != 0 && ans[len(ans) - 1] == popped[j] {
      j++
      ans = ans[:len(ans) - 1]
    }
  }
  return len(ans) == 0
}

File: LeetCode/DataStruct/Rust/Leetcode946-validate-stack-sequences.rs
impl Solution {
  pub fn validate_stack_sequences(pushed: Vec<i32>, popped: Vec<i32>) -> bool {
    let (mut ans, n) = (vec![], pushed.len());
    let mut j = 0;
    for i in 0..n {
      ans.push(pushed[i]);
      while !ans.is_empty() && ans[ans.len() - 1] == popped[j] {
        j += 1;
        ans.pop();
      }
    }
    ans.is_empty()
  }
}
